### PR TITLE
Changes to derive Asset base from the Asset identifier

### PR DESCRIPTION
**Pull request by vivek-arte**
This adds the changes to the zcash test vectors to update the calculation of the Asset Base from the Asset Identifier via the Asset Digest, as detailed in ZIP 227.

### CODE REVIEW

File: zcash_test_vectors/orchard/asset_id.py [hunk 1]
```diff
@@ -5,15 +5,28 @@
...
+    return b"\x00" + key + description
+
+
+def asset_digest(encoded_asset_id):
+    digest = blake2b(person=b"ZSA-Asset-Digest")
+    digest.update(encoded_asset_id)
+    return digest.digest()
+
+
+def asset_base(asset_digest_value):
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/asset_id.py:27]: let's use `ZSAValueBase()` from the zip (with proper case).

File: zcash_test_vectors/orchard/asset_id.py [hunk 2]
```diff
@@ -70,12 +83,12 @@ def randbytes(l):
 
         key_bytes = bytes(fvk.ivk())
         description_bytes = get_random_unicode_bytes(512)
-        asset = asset_id(key_bytes, description_bytes)
+        asset = asset_base(asset_digest(encode_asset_id(key_bytes, description_bytes)))
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/asset_id.py:86]: The naming here and below is inconsistent. should be
```python
asset_digest = asset_digest(encode_asset_id(key_bytes, description_bytes))
asset_base = asset_base(asset_digest)
```

File: zcash_test_vectors/orchard/asset_id.py [hunk 3]
```diff
@@ -70,12 +83,12 @@ def randbytes(l):
...
 
         test_vectors.append({
             'key': key_bytes,
             'description': description_bytes,
-            'asset_id': bytes(asset),
+            'asset_base': bytes(asset),
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/asset_id.py:91]: ` 'asset_base': bytes(asset_base),`

File: zcash_test_vectors/orchard/asset_id.py [hunk 4]
```diff
@@ -5,15 +5,28 @@
...
 
 
 def native_asset():
     return group_hash(b"z.cash:Orchard-cv", b"v")
 
-def asset_id(key, description):
-    return group_hash(b"z.cash:Orchard-cv", key + description)
+
+def encode_asset_id(key, description):
+    return b"\x00" + key + description
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/asset_id.py:18]: separate `\x00` and call it by name (from the zip: version byte)

File: zcash_test_vectors/orchard/asset_id.py [hunk 5]
```diff
@@ -5,15 +5,28 @@
...
+
+
+def asset_digest(encoded_asset_id):
+    digest = blake2b(person=b"ZSA-Asset-Digest")
+    digest.update(encoded_asset_id)
+    return digest.digest()
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/asset_id.py:24]: I don't like `digest.digest()`. Let's rename `digest` to `h`.

**Review by PaulLaux — APPROVED** [review]
Very nice, approved pending comments.

In addition, add a proper description to the PR.

When merging, make sure to copy the updated PR description to the merge comment box. This is done to have a proper `git commit` msg.

File: zcash_test_vectors/orchard/asset_id.py (hunk 1 above)
Comment (vivek-arte) [reply zcash_test_vectors/orchard/asset_id.py:27]: Switched to `zsa_value_base()`.

File: zcash_test_vectors/orchard/asset_id.py (hunk 5 above)
Comment (vivek-arte) [reply zcash_test_vectors/orchard/asset_id.py:24]: Done.

File: zcash_test_vectors/orchard/asset_id.py (hunk 2 above)
Comment (vivek-arte) [reply zcash_test_vectors/orchard/asset_id.py]: Changed it to `asset_base = asset_base(asset_digest(encode_asset_id(key_bytes, description_bytes)))`

File: zcash_test_vectors/orchard/asset_id.py (hunk 4 above)
Comment (vivek-arte) [reply zcash_test_vectors/orchard/asset_id.py]: Done.

File: zcash_test_vectors/orchard/asset_id.py (hunk 3 above)
Comment (vivek-arte) [reply zcash_test_vectors/orchard/asset_id.py]: Done

**Pull request merged**